vent/api/actions.py

```diff
@@ -79,221 +83,272 @@ def prep_start(self,
+
+                # add label for priority
+                status = vent_template.section('settings')
+                self.logger.info(status)
+                if status[0]:
+                    for option in status[1]:
+                        if option[0] == 'priority':
+                            tool_dict[container_name]['labels']['vent.priority'] = option[1]
+
+                # only start tools that have been built
+                if sections[section]['built'] != 'yes':
+                    del tool_dict[container_name]
+
+                self.logger.info(tool_dict)
+
+            # check and update links, volumes_from, network_mode
+            for container in tool_dict.keys():
+                if 'links' in tool_dict[container]:
+                    for link in tool_dict[container]['links']:
+                        for c in tool_dict.keys():
+                            if 'tmp_name' in tool_dict[c] and tool_dict[c]['tmp_name'] == link:
+                                tool_dict[container]['links'][tool_dict[c]['name']] = tool_dict[container]['links'].pop(link)
+                if 'volumes_from' in tool_dict[container]:
+                    tmp_volumes_from = tool_dict[container]['volumes_from']
+                    tool_dict[container]['volumes_from'] = []
+                    for volumes_from in list(tmp_volumes_from):
+                        for c in tool_dict.keys():
+                            if 'tmp_name' in tool_dict[c] and tool_dict[c]['tmp_name'] == volumes_from:
+                                tool_dict[container]['volumes_from'].append(tool_dict[c]['name'])
+                                tmp_volumes_from.remove(volumes_from)
+                    tool_dict[container]['volumes_from'] += tmp_volumes_from
+                if 'network_mode' in tool_dict[container]:
+                    if tool_dict[container]['network_mode'].startswith('container:'):
+                        network_c_name = tool_dict[container]['network_mode'].split('container:')[1]
+                        for c in tool_dict.keys():
+                            if 'tmp_name' in tool_dict[c] and tool_dict[c]['tmp_name'] == network_c_name:
+                                tool_dict[container]['network_mode'] = 'container:'+tool_dict[c]['name']
+
+            self.logger.info(tool_dict)
+
+            # remove tmp_names
+            for c in tool_dict.keys():
+                if 'tmp_name' in tool_dict[c]:
+                    del tool_dict[c]['tmp_name']
+
+            # remove containers that shouldn't be started
+            for c in tool_dict.keys():
+                if 'start' in tool_dict[c] and not tool_dict[c]['start']:
+                    del tool_dict[c]
+
+            self.logger.info(tool_dict)
+        except Exception as e:
+            self.logger.error(str(e))
+            status = (False, e)
+
+        status = (True, tool_dict)
+        self.logger.info(status)
+        self.logger.info("Finished: prep_start")
+        return status
```

Comment:
I think I've modified start accordingly so that it takes status[1] to get tool_dict for the update that it performs.